### PR TITLE
Null in index scan checking

### PR DESCRIPTION
**Pull request by tjoneslo**
Address one more case of using NULL in statements which use an index scan incorrectly. In this case `SELECT * FROM t WHERE x = ?` where the parameter is set to `null` should return no rows, but the query  `SELECT * from t WHERE x IS NULL;` should return rows. In `IndexCursorUnidirectional.java`, there isn't any way to distinguish between these two cases. This branch adds one. 

It checks the underlying implementations of the expressions used for the range on the scan to ensure the source is a literal null or not. If so, the second query is assumed and does the range correctly. If not a literal, the first form is assumed, and the cursor is flagged to not return any rows. 

To make the IndexCursorUnidirectional as aggressive about skipping the processing, including creating the underlying FDBIterator and K/Vstore scan, it uses the cursor state Idle setting. Because the implementation for this case is aggressive (to avoid the expensive transaction processing), the `IndexCursorUnidirectional#jump()` cursor processing needs to reset the cursor from idle to active to allow correct processing.

### CODE REVIEW

**Comment by MMcM** [comment]
I'm a little concerned that the new `isLiteralNull` returns `true` for non-null literals because it is only asked after a null value check. There is no problem in this branch, but I see a potential for confusion later.

**Comment by tjoneslo** [comment]
Renamed.

File: fdb-sql-layer-core/src/main/java/com/foundationdb/qp/expression/RowBasedUnboundExpressions.java [hunk 1]
```diff
@@ -29,10 +29,15 @@
 import com.foundationdb.server.explain.Type;
 import com.foundationdb.server.types.value.ValueRecord;
 import com.foundationdb.server.types.value.ValueSource;
+import com.foundationdb.server.types.texpressions.TNullExpression;
 import com.foundationdb.server.types.texpressions.TPreparedExpression;
+import com.foundationdb.server.types.texpressions.TPreparedLiteral;
```
Comment (MMcM) [inline fdb-sql-layer-core/src/main/java/com/foundationdb/qp/expression/RowBasedUnboundExpressions.java]: New imports do not appear to be used in the end.

**Comment by MMcM** [comment]
Substantive changes look right to me. Cursor lifecycle is always delicate, but I can't find any problem.
Ready to go I think after deciding whether or not to clean up imports.

**Comment by tjoneslo** [comment]
removed unused imports

**Comment by MMcM** [comment]
Thanks.
Approve.

**Pull request merged**